src/request-pipeline/index.js

```diff
@@ -113,7 +113,11 @@ function createReqOpts (ctx) {
 function sendResponseHeaders (ctx) {
     var headers = headerTransforms.forResponse(ctx);
 
-    ctx.res.writeHead(ctx.destRes.statusCode, headers);
+    if (ctx.contentInfo.requireProcessing && ctx.destRes.statusCode === 204 && headers['content-length'] > 0)
```

Comment:
Wouldn't it be better to not process content with 204 status at all? BTW, how browser behave if the receive 204 for the page? Do they perform page transfer? According to spec, browser should just updated meta information from headers. If it's shows blank page, then we can signal it as fatal error for the page, but otherwise we should keep things as is.